File: src/ipc/launchOptions.ts
import { invoke } from "./invoke";
import { LaunchOptionCatalog, LaunchOptionSelection, LaunchWarning } from "./types";

export const getLaunchOptionCatalog = () =>
  invoke<LaunchOptionCatalog>("get_launch_option_catalog");

export const validateLaunchArgs = (selection: LaunchOptionSelection) =>
  invoke<LaunchWarning[]>("validate_launch_args_cmd", { selection });

export const composeLaunchArgs = (selection: LaunchOptionSelection) =>
  invoke<string[]>("compose_launch_args_cmd", { selection });

File: src-tauri/src/commands/launch_options.rs
use crate::error::AppResult;
use crate::launch_options::{
    catalog, compose_launch_args, validate_launch_args, LaunchOptionCatalog, LaunchOptionSelection,
    LaunchWarning,
};

#[tauri::command]
pub fn get_launch_option_catalog() -> AppResult<&'static LaunchOptionCatalog> {
    Ok(catalog())
}

#[tauri::command]
pub fn validate_launch_args_cmd(selection: LaunchOptionSelection) -> AppResult<Vec<LaunchWarning>> {
    Ok(validate_launch_args(&selection))
}

#[tauri::command]
pub fn compose_launch_args_cmd(selection: LaunchOptionSelection) -> AppResult<Vec<String>> {
    Ok(compose_launch_args(&selection))
}
